pkg/fpga/ioctl_linux.go

```diff
@@ -21,13 +21,15 @@ import (
 
 func ioctl(fd uintptr, req uint, arg uintptr) (ret uintptr, err error) {
 	ret, _, en := syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
+	// Even though en is syscall.Errno which implements the error interface it's
+	// an unsigned number and can't be nil. So filter the zero value out.
 	if en != 0 {
-		err = syscall.Errno(en)
+		err = en
 	}
 	return
```

Comment:
should we get rid of naked return to avoid this type of errors in future?
I believe this code looks much more understandable than the original:
```
func ioctl(fd uintptr, req uint, arg uintptr) (uintptr, error) {
        ret, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
        if err != 0 {
                err = syscall.Errno(en)
        }
        return ret, err
}
```

Reply:
Golang's doc suggests what @rojkov has: https://golang.org/pkg/syscall/#Errno

Reply:
ok, makes sense then thank you for the explanations!

Reply:
anyway, there is something implicit I don't like in this code. How about this one?
```
func ioctl(fd uintptr, req uint, arg uintptr) (uintptr, error) {
        ret, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
        if err == 0 {
                return ret, nil
        }
        return ret, err
}
```

Reply:
Naked returns are not the reason for this mistake. The reason is that we are too accustomed to
```go
if err != nil {
    ...
}
```
checks. It turned out an object implementing the `error` interface can be not `nil`, yet not indicating an error condition, but success.

Though I agree that explicit return values everywhere are more readable. But the nakedret check is ok with this code.

Reply:
Oops, didn't notice your reply. Ok, will update.